json-schema/json-schema/src/main/scala/endpoints/algebra/JsonSchemas.scala

```diff
@@ -82,6 +82,9 @@ trait JsonSchemas {
   /** The JSON schema of a record with a single optional field `name` of type `A` */
   def optField[A](name: String, documentation: Option[String] = None)(implicit tpe: JsonSchema[A]): Record[Option[A]]
 
+  /** Name of the field that acts as discriminator for sum types (coproducts) */
+  def discriminatorName: String = "type"
```

Comment:
For sake of flexibility, it might be preferable to make `discriminatorName` an additional parameter of `def tagged` rather than an algebra-wise value, so that the users are able to specify different discriminator names like `pet_type` in the corresponding example in OAS3 document.
 
![image](https://user-images.githubusercontent.com/969120/46621376-47678400-cb62-11e8-97c0-840b585b3b0d.png)

Reply:
oops, `taggedRecord` I mean...

Reply:
See https://github.com/julienrf/endpoints/pull/135#discussion_r223656666 which concerns the same issue.